src/logic/two_pointed.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.sum.basic
+import logic.nontrivial
+
+/-!
+# Two-pointed types
+
+This file defines `two_pointed`, a class holding a two-pointing of a type.
+
+This is morally a Type-valued `nontrivial`.
+-/
+
+open function
+
+variables {α β : Type*}
+
+/-- Two-pointing of a type. Type-valued `nontrivial`. -/
+class two_pointed (α : Type*) :=
+(fst snd : α)
+(fst_ne_snd : fst ≠ snd)
```

Comment:
When you say it's morally a Type-valued `nontrivial`, there are two ways that could happen I think: either an ordered pair or an unordered pair, though ordered pair is more likely.  What's the application for this class?

Do you think this should be generalized to n-pointed? Something like
```lean
class n_pointed (α : Type*) (n : ℕ) :=
(pts : list α)
(pts_length : pts.length = n)
```
Then you could have
```lean
abbreviation two_pointed (α : Type*) := n_pointed α 2
```

Reply:
This is what I'm trying to formalize: http://nlab-pages.s3.us-east-2.amazonaws.com/nlab/show/coalgebra+of+the+real+interval
So I think it shouldn't be generalized, at least not here. Also, you forgot `nodup`.